DBZ-3269 DBZ-3280 Scripting docs improvements (#2218)
debezium/debezium: documentation/modules/ROOT/pages/configuration/filtering.adoc

Expressions should not result in any side-effects. That is, they should not modify any variables that they pass.

[NOTE]
====

> {prodname} emits not only data messages but also metadata messages about schema chages, transactions or heartbeat messages.
> The expressions should not be applied on such messages as their structure is different from what is usually expected by the expression.
> It is recommended to use either xref:filter-topic-regex[topic.regex] configuration option or a link:https://cwiki.apache.org/confluence/display/KAFKA/KIP-585%3A+Filter+and+Conditional+SMTs[an SMT predicate] to ensure that the expression is applied only to a valid messages.

Reviewer: ```suggestion
{prodname} emits not only data change events, but also metadata messages about schema changes, transactions, or heartbeat messages.
Typically, filter expressions should not be applied to such messages, as their structure is different from what is usually expected by the expression.
It is recommended to use either the xref:filter-topic-regex[topic.regex] configuration option or an link:https://cwiki.apache.org/confluence/display/KAFKA/KIP-585%3A+Filter+and+Conditional+SMTs[SMT predicate] to ensure that the expression is applied only to data change events.
```
